### PR TITLE
Final Changes with Dr. J notes

### PR DESCRIPTION
**Pull request by Kahjax**

### CODE REVIEW

File: KJ1124/Thor_Puzzle_assign.c [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    // the Y coordinate position of the light of power
+    int light_y; 
+    // Thor's starting position at X
+    int initial_tx; 
+    // Thor's starting position at Y
+    int initial_ty;
+    // The cin.ignore() function is used to ignore characters that may not belong in your input buffer
+    cin >> light_x >> light_y >> initial_tx >> initial_ty; cin.ignore();
+
+/* This while statement is the beginning of the beginning of the game loop and it allows causes all of your test cases to run
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Make this a `//` comment and indent it to align with the `while` statement.

File: KJ1124/Thor_Puzzle_assign.c [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        cin >> remaining_turns; cin.ignore();
+
+/*
+Diagram
+---------
+
+.. image:: ThorPwr.jpg
+
+
+*/
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Combine these two comments -- there's no reason to end one then just begin another.

File: KJ1124/Thor_Puzzle_assign.c [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+*       N (North) 
+*       NE (North East) 
+*       E (East)
+*       SE (South East) 
+*       S (South)
+*       SW (South West) 
+*       W (West)
+*       NW (North West)
+*/
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Combine comments

File: KJ1124/Thor_Puzzle_assign.c [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+/* 
+Thor's Movement
+----------------
+*/
+
+        /* In this case the light of power at the X and Y coordinates are greater than Thor's initial condition so 2 if condition statements have to be implemented. Thor's initial X and Y conditions must be incremented so he can move in a diagonal direction without going out of bounds or repositioning from the light of power.
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Use `//` for single-line comments, here and throughout the file.

Comment (bjones1) [reply KJ1124/Thor_Puzzle_assign.c]: Why does `(light_y > initial_ty) && (light_x > initial_tx)` means move SE? Why not NW or W or something else? Explain.

File: KJ1124/Thor_Puzzle_assign.c [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        */
+        if ((light_y > initial_ty) && (light_x > initial_tx))
+        {
+            cout << "SE" << endl;
+            initial_tx++;
+            initial_ty++;
+            }
+        /* Based on the compass it can be determined that Thor will travel E (East).The light_x is the position of the light of power  and initial_tx is Thor's starting position in the X-coordinate plane. The postion of the light of power is is going in the positive direction and it greater than Thor's starting postion so a condition statement is created that if this is the case travel East.
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Why is `initial_tx` not updated? The same question applies for the rest of your if statements.

Comment (bjones1) [reply KJ1124/Thor_Puzzle_assign.c]: Why does `light_x > initial_tx` mean move east?

File: KJ1124/Thor_Puzzle_assign.c [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        */
+        else if (light_x > initial_tx) {
+            cout << "E" << endl;
+            }
+        /* Thor will travel N (North) since the light of power is above his location and it is in the positive direction .The light_y is the position of the light of power and initial_ty is Thor's starting position in the Y-coordinate plane.
+        */
+        else if (light_y < initial_ty) {
+            cout << "N" << endl;
+            }
+        /* In this instance the light of power is in the negative Y direction of Thor's position. It is also at an angle which means Thor must move in an diagonal direction so that he does not run out of energy. He will travel diagional because of the increment then straight.
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Why only increment one variable? Why only test on variable for a diagonal move?

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Comment by bjones1** [comment]
I haven't seen any progress on this. Message me after you push update and are ready for another review.

**Comment by Kahjax** [comment]
Ready for final changes review

File: KJ1124/Thor_Puzzle_assign.c [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+*       E (East)
+*       SE (South East) 
+*       S (South)
+*       SW (South West) 
+*       W (West)
+*       NW (North West)
+ 
+Thor's Movement
+----------------
+.. image:: XYgraph.jpg
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c:82]: Starting here, indent your comments so they align with your code below.

File: KJ1124/Thor_Puzzle_assign.c [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+In this case the light of power at the X and Y coordinates are greater than Thor's initial condition so, two if condition statements must to be implemented. Futhermore, the Y coordinate values increase as you go South and the X coordinates vale increase as you go further East. Thor's initial X and Y conditions must be incremented so he can move in the South East without going pass the MAX boundary for both the X and Y parameters.
+
+        */
+        if ((light_y > initial_thor_y) && (light_x > initial_thor_x))
+        {
+            cout << "SE" << endl;
+            initial_thor_x++;
+            initial_thor_y++;
+            }
+/* Based on the compass it can be determined that Thor will travel E (East).The light_x is the position of the light of power  and initial_thor_x is Thor's starting position in the X-coordinate plane. The postion of the light of power is is going in the positive direction and it greater than Thor's starting postion so a condition statement is created that if this is the case travel East.
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c]: Indent (here and following) to align with code.

File: KJ1124/Thor_Puzzle_assign.c [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        */
+        else if (light_y < initial_thor_y) {
+            cout << "N" << endl;
+            initial_thor_y--;
+            }
+/* Based on the if statement for the South East direction, a comparison is made testing whether my y-value and x-value of the light power is greater than Thor's position, based on that outcome the code will go through the else-if statements.If both conditions apply, then SW would be the direction that Thor will move and this is the reason for the one variable increment.He will travel diagional because of the increment then straight.
+        */
+        else if (light_y > initial_thor_y) {
+            cout << "SW" << endl;
+            initial_thor_y++;
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c:115]: Why does x not change here?

File: KJ1124/Thor_Puzzle_assign.c [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+            // *Positive increment here so that Thor does not run out of bounds.*
+            }
+        // This else if statement is so that Thor will travel West after incrementing to the point where the light of power is equal to Thor's initial position.
+        else if (light_x < initial_thor_x) {
+            cout << "W" << endl;
+            }
+    }
+} 
+// **ending bracket**
```
Comment (bjones1) [inline KJ1124/Thor_Puzzle_assign.c:124]: What about the other directions (NW, S, etc)?

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Comment by bjones1** [comment]
These comments still need to be addressed...

**Comment by bjones1** [comment]
Ping...still waiting on this...

**Comment by Kahjax** [comment]
Dr. Jones,

I haven’t pushed the changes I made because I have not had good service on my computer on the trip. I will make these changes as a soon as I can.

Thanks in advance!

Kaitlyn Jackson
Electrical Engineering
Mississippi State University
(228)990-455
________________________________

**Comment by bjones1** [comment]
I look forward to your updates. Comment here after you push so I know that you're ready for review.

**Comment by Kahjax** [comment]
I made the necessary changes I forgot to comment here. Let me know if you cant see it

**Comment by bjones1** [comment]
I can't see any new commits.

**Pull request merged**